pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1031,6 +1083,30 @@ func doActivate(ctx *domainContext, config types.DomainConfig,
 			ds.FileLocation, ds.ActiveFileLocation)
 	}
 
+	// Only in Add case
+	// Check whether a container image
+	// if (status.PendingAdd) && (config.DiskConfigList[0].Format == "container") {
+	// Looks like the value is 8 stringified and not a enum nor "container"- To Be Fixed
+	if (status.PendingAdd) && (config.DiskConfigList[0].Format == "8") {
```

Comment:
Ok, speaking of re-structuring code -- I suggest we fold as much of it as possible into the general flow of xlCreate function. In fact, why don't we rename and parameterize xlCreate to take care of either xl or rkt entry points for launching a domain?

You'll have to move that call there anyway -- since you'll have to have xl.cfg ready to be passed to rkt.

Makes sense?

Reply:
I agree with keeping as much code as common as possible. Will try your suggestion of renaming xlcreate and branching out based on container or not.